Lib/functools.py

```diff
@@ -868,3 +868,41 @@ def _method(*args, **kwargs):
     @property
     def __isabstractmethod__(self):
         return getattr(self.func, '__isabstractmethod__', False)
+
+
+################################################################################
+### cached_property() - computed once per instance, cached as attribute
+################################################################################
+
+_NOT_FOUND = object()
+
+
+class cached_property:
+    def __init__(self, func):
+        self.func = func
+        self.attrname = func.__name__
+        self.__doc__ = func.__doc__
+        self.lock = RLock()
+
+    def __set_name__(self, owner, name):
+        self.attrname = name
```

Comment:
Might make sense to raise error in this case, since `a` will never be cached:
```
class C:
    @cached_property
    def a(self):
        pass
    b = a
```

Reply:
OK. I guess that also answers the above question; we will raise error in case `__set_name__` is ever called twice, and also I guess raise error if `__get__` is called and `__set_name__` was not.